Updates and Reformating

KaNaDaAT opened the pull request:
- Arrays now have a keyword `:len` for length
- Arrays now can be used with variables as indexer
- _true_ and _false_ now work properly
- _funqtions_ can now return in the middle of them self and not jsut at the end
- Now supports `~?` as alias for _if_
- No longer has issues with variables from "higer" contexts
- _funqtions_ without parameters possible now
- _string_ can also be decleared empty now
- **Lib:** Better Console
- **Lib:** Random Number
- **Lib:** Parse String to int/decimal/double/bool
- Some Bugs Fixed

Commit:
Updates and Reformating

- Arrays now have a keyword `:len` for length
- Arrays now can be used with variables as indexer
- _true_ and _false_ now work properly
- _funqtions_ can now return in the middle of them self and not jsut at the end
- Now supports `~?` as alias for _if_
- No longer has issues with variables from "higer" contexts
- _funqtions_ without parameters possible now
- _string_ can also be decleared empty now
- **Lib:** Better Console
- **Lib:** Random Number
- **Lib:** Parse String to int/decimal/double/bool
- Some Bugs Fixed

qrakhen commented on sqripts/loops.sq:
```diff
@@ -1,7 +1,6 @@
 	n <~ n + 1;
-	*~ t <~ "we're at step " + n + "
-	";
+	*~ t <~ "we're at step " + n + "\n";
```
didn't know that worked yet, lol

KaNaDaAT replied:
Cause I added it :P haha

Commit:
Refactoring and "!=" support.

Commit:
Refactoring

Commit:
ToDo and new console:write functionality

Commit:
Another Roudn of Refactoring

qrakhen commented on Core/Digesters/Funqtionizer.cs:
```diff
@@ -2,62 +2,84 @@
-    {
-        public FunqtionizerException(string message, Token cause = null) : base(message, cause) { }
-    }
+
```
```suggestion
```

qrakhen commented on sqripts/Testing.sq:
```diff
@@ -0,0 +1,225 @@
+	~? ((input != 'y') && (input != 'Y')) {
+		run <~ 0;
+	}
+}
```
i don't think we need the testing sqript here, but i still appreciate the contribution :D

KaNaDaAT replied:
Ah yeah the Testing.sq was the one executing automatically in VS as args param. You can remove that.

qrakhen commented on sqripts/backup.sq:
```diff
@@ -0,0 +1,18 @@
+	}
+});
+
+printList(emptyboard);
```
what?! [[1],[2]] works? holy shit!

Commit:
Fixed "backup.sq"

Renamed `Backup.sq` to  `IterateArrayWithIndex.sq`.
Forgot a line of code. Fixed now.
